crypto/dh/dh_check.c

```diff
@@ -139,21 +137,7 @@ int DH_check(const DH *dh, int *ret)
             *ret |= DH_CHECK_INVALID_Q_VALUE;
         if (dh->j && BN_cmp(dh->j, t1))
             *ret |= DH_CHECK_INVALID_J_VALUE;
-
-    } else if (BN_is_word(dh->g, DH_GENERATOR_2)) {
-        l = BN_mod_word(dh->p, 24);
-        if (l == (BN_ULONG)-1)
-            goto err;
-        if (l != 11)
-            *ret |= DH_NOT_SUITABLE_GENERATOR;
```

Comment:
We have tests in Node.js that depended on this block of removed code. Perhaps wrongly. See https://github.com/nodejs/node/pull/29550#issuecomment-531542504 for a discussion of what we could do work around this change as we update to 1.1.1d. Is it the position of OpenSSL that mods of 11 and 7 are in fact suitable generators, and that no one using DH with this param should have/would have been paying attention to this rc?

Reply:
You need a safe prime of at least 1024 bits, or better 2048 that will be safe for 30+ years.
g can generate the order-q or 2q subgroup but must of course not be 0, 1, p-1.

So 11 and 7 are safe primes, but they will be rejected in not too far future.

The order-q subgroup is preferable but even if we stop generating 2q subgroups
we will not have a check for g being a quadratic residue for backward compatibility.

For good tests vectors I would suggest use p=2048-bit safe prime,
and g=quadratic non-residue (as generated previously)

For bad test vectors use g=p-1, or p=not prime, or not safe prime.

Reply:
Thanks for the background, @bernd-edlinger